datasets.py

```diff
@@ -919,3 +920,110 @@ def from_path(self, file_path):
+        # (e.g. aachen/aachen_000000_000019_gtFine_color.png = aachen/aachen_000000_000019)
+        self.image_ids = []
+        for city in self.city_names:
+            filename_path = os.path.join(self.gps_folder, city)
+            city_filenames = os.listdir(filename_path)
+            self.image_ids = self.image_ids +  [os.path.join(city, name.split("_vehicle")[0]) for name in city_filenames]
+        print("done with ids (1/2)")
+
+        with open("/Users/home/Downloads/stanford-nh891yz3147-geojson.json") as f:
+            self.geo_boundaries = json.load(f)
+
+        # TODO: categories work
+        lab_arr = ['unlabeled',
+            'ego vehicle',
+            'rectification border',
+            'out of roi',
+            'static',
+            'dynamic',
+            'ground',
+            'road',
+            'sidewalk',
+            'parking',
+            'rail track',
+            'building',
+            'wall',
+            'fence',
+            'guard rail',
+            'bridge',
+            'tunnel',
+            'pole',
+            'polegroup',
+            'traffic light',
+            'traffic sign',
+            'vegetation',
+            'terrain',
+            'sky',
+            'person',
+            'rider',
+            'car',
+            'truck',
+            'bus',
+            'caravan',
+            'trailer',
+            'train',
+            'motorcycle',
+            'bicycle',
+            'license plate']
+        self.labels_to_names = {i : i for i in lab_arr}
+
+        print("done with categories (2/2)")
+
+
+
+    def __getitem__(self, index):
+        image_id = self.image_ids[index]
+        return self.from_path(image_id)
+    
+    def __len__(self):
+        return len(self.image_ids)
+    
+    def from_path(self, file_path):
+        image_id = os.path.join(self.img_folder, "{0}_gtFine_color.png".format(file_path))
+        image = Image.open(image_id).convert("RGB")
+        # image_size = list(image.size())[1:]
+
+        country = None
+
+        # for each image, get category information
+        image_anns = []
+
+        category_path = os.path.join(self.img_folder, "{0}_gtFine_polygons.json".format(file_path))
+        category_data = json.load(open(category_path)).get('objects', [])
+
+        
+        for i in range(len(category_data)):
+            label = category_data[i].get('label', None)
+            if label is not None:
+                image_anns.append({'label': label})
+
+        # for each image, get long lat gps information
+        lat_lng = {}
+        json_data = json.load(open(os.path.join(self.gps_folder, "{0}_vehicle.json".format(file_path))))
+        if json_data is not None and "gpsLatitude" in json_data and "gpsLongitude" in json_data:
+            lat_lng['lat'] = json_data["gpsLatitude"]
+            lat_lng['lng'] = json_data["gpsLongitude"]
+
+        anns = [image_anns, None, [country], file_path, None, lat_lng]    
```

Comment:
can you add into the template dataset what this new lat_lng you're adding is, and what format it's expected?

Reply:
added